lib/Compilation.js

```diff
@@ -1779,17 +1782,26 @@ class Compilation extends Tapable {
 
 		// PART TWO
 		/** @type {Set<Module>} */
-		let availableModules;
-		/** @type {Set<Module>} */
 		let newAvailableModules;
-		/** @type {Queue<AvailableModulesChunkGroupMapping>} */
-		const queue2 = new Queue(
-			inputChunkGroups.map(chunkGroup => ({
-				chunkGroup,
-				availableModules: new Set(),
-				needCopy: true
-			}))
-		);
+
+		/**
+		 * @typedef {Object} ChunkGroupInfo
+		 * @property {Set<Module>} minAvailableModules current minimal set of modules available at this point
+		 * @property {Set<Module>[]} availableModulesToBeMerged enqueued updates to the minimal set of available modules
+		 */
+
+		/** @type {Map<ChunkGroup, ChunkGroupInfo>} */
+		const chunkGroupInfoMap = new Map();
+
+		/** @type {Queue<ChunkGroup>} */
+		const queue2 = new Queue(inputChunkGroups);
+
+		for (const chunkGroup of inputChunkGroups) {
+			chunkGroupInfoMap.set(chunkGroup, {
+				minAvailableModules: undefined,
+				availableModulesToBeMerged: [new Set()]
```

Comment:
Why does the array contain a `Set` and not on line 1912?

Reply:
This handles the entrypoints which start with "no modules available".

line 1912 handles non-entrypoints which start uninitialized and available modules are inherited from parents.